### PR TITLE
CB2-9315: Generate HGV PRS Bilingual

### PR DESCRIPTION
**Pull request by cb-cs**
## Description

Related issue: [CB2-9315](https://dvsa.atlassian.net/browse/CB2-9315)

## Before submitting (or marking as "ready for review")

- [ ] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ ] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

**Review by naathanbrown — APPROVED** [review]

File: src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        );
+        assertEquals(
+                String.format("%s%s%s %s", "(", EU_NUMBER_SUMMARY_HEADER, ")", FAILED_SUMMARY_HEADER_WELSH),
+                resultNameFailWelsh
+        );
+    }
+
+    @Test
+    public void verifyVin() {
+        String vin = certificatePageObjectVTG5.getVin();
```
Comment (Daniel-Searle) [inline src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java:76]: It's probably better for these to be named vinPass to make it clear vs vinFail

File: src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+        assertEquals(testCertificate.getData().getFormattedCurrentOdometer(), mileage);
+        assertEquals(testCertificate.getFailData().getFormattedCurrentOdometer(), mileageFail);
+        assertEquals("20,000 cilometrau", mileageWelsh);
+        assertEquals("20,000 cilometrau", mileageFailWelsh);
+    }
+
+    @Test
+    public void verifyPrsDefects() {
+        //
```
Comment (Daniel-Searle) [inline src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java:153]: test's empty

File: src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        String orgNameFail = certificatePageObjectVTG30.getTestingOrganisationAndInspectorsName();
+        String orgNameWelsh = certificatePageObjectVTG5W.getTestingOrganisationAndInspectorsName();
+        String orgNameFailWelsh = certificatePageObjectVTG30W.getTestingOrganisationAndInspectorsName();
+
+        String organisation = testCertificate.getData().getTestingOrganisation();
+        String organisationFail = testCertificate.getFailData().getTestingOrganisation();
+        String testerName = testCertificate.getData().getIssuersName();
+        String testerNameFail = testCertificate.getFailData().getIssuersName();
+
+        String organisationWelsh = testCertificate.getFailData().getTestingOrganisationWelsh();
```
Comment (Daniel-Searle) [inline src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java:200]:
```suggestion
              String organisation = testCertificate.getData().getTestingOrganisation();
        String organisationFail = testCertificate.getFailData().getTestingOrganisation();
        String organisationWelsh = testCertificate.getFailData().getTestingOrganisationWelsh();
        
        String testerName = testCertificate.getData().getIssuersName();
        String testerNameFail = testCertificate.getFailData().getIssuersName();
```

File: src/test/java/pdfverification/tests/HgvPRSBilingualTests.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    private static final String CERT_NAME_FAIL = "Refusal of MOT test certificate";
+    private static final String CERT_NAME_WELSH = "Tystysgrif prawf MOT (HGV)";
+    private static final String CERT_NAME_FAIL_WELSH = "Gwrthod tystysgrif prawf MOT";
+
+    private HtmlGenerator htmlGenerator;
+    private PDFGenerationService pdfGenerationService;
+    private CvsMotCertificate testCertificate;
+    private PDFParser pdfParser;
+    private PdfReader pdfReader;
+    private byte[] pdfData;
```
Comment (Daniel-Searle) [inline src/test/java/pdfverification/tests/HgvPRSBilingualTests.java:29]:
```suggestion
     private final HtmlGenerator htmlGenerator;
    private final PDFGenerationService pdfGenerationService;
    private CvsMotCertificate testCertificate;
    private final PDFParser pdfParser;
    private PdfReader pdfReader;
```

File: src/test/java/pdfverification/tests/HgvPRSBilingualTests.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        this.htmlGenerator = new HtmlGenerator(new Handlebars());
+        this.pdfGenerationService = new PDFGenerationService(new ITextRenderer());
+        this.pdfParser = new PDFParser();
+    }
+
+    @Before
+    public void setup() throws Exception {
+        pdfData = pdfGenerationService.generate(htmlGenerator.generate(testCertificate));
+        pdfReader = pdfParser.readPdf(pdfData);
+    }
```
Comment (Daniel-Searle) [inline src/test/java/pdfverification/tests/HgvPRSBilingualTests.java:42]:
```suggestion
    @Before
    public void setup() throws Exception {
        byte[] pdfData = pdfGenerationService.generate(htmlGenerator.generate(testCertificate));
        pdfReader = pdfParser.readPdf(pdfData);
    }
```

File: src/test/java/pdfverification/tests/HgvPRSBilingualTests.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import uk.gov.dvsa.model.cvs.CvsMotCertificate;
+import uk.gov.dvsa.service.HtmlGenerator;
+import uk.gov.dvsa.service.PDFGenerationService;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+
+public class HgvPRSBilingualTests {
+    private static final String CERT_NAME = "MOT test certificate (HGV)";
```
Comment (Daniel-Searle) [inline src/test/java/pdfverification/tests/HgvPRSBilingualTests.java:19]: IT's probably not worth making these variables, they're only being used in one place currently

File: src/test/java/htmlverification/service/CvsCertificateTestDataProvider.java [hunk 7]
```diff
@@ -679,7 +679,78 @@ public static CvsHgvTrlFailBilingual getCvsHgvTrlFailBilingual() {
...
+                        )
+                )
+                .setIssuersName("TESTER NAME")
+                .setTestStationName("TEST STATION NAME")
+                .setTestNumber("X01X00001")
+                .setEarliestDateOfTheNextTest("01.07.2024")
+                .setTrn(null)
+                .setIsTrailer(false);
+
+        bilingualFailData
```
[the remark this answers is not in the document]
Comment (cb-cs) [reply src/test/java/htmlverification/service/CvsCertificateTestDataProvider.java:720]: updated

File: src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java (hunk 1 above)
Comment (cb-cs) [reply src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java:76]: updated

File: src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java (hunk 2 above)
Comment (cb-cs) [reply src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java:153]: updated

File: src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java (hunk 3 above)
Comment (cb-cs) [reply src/test/java/htmlverification/tests/CvsHgvPRSBilingualTest.java:200]: done

File: src/test/java/pdfverification/tests/HgvPRSBilingualTests.java (hunk 4 above)
Comment (cb-cs) [reply src/test/java/pdfverification/tests/HgvPRSBilingualTests.java:29]: Not sure what this suggested change is??

Comment (cb-cs) [reply src/test/java/pdfverification/tests/HgvPRSBilingualTests.java:29]: Remove the variable and use locally?

File: src/test/java/pdfverification/tests/HgvPRSBilingualTests.java (hunk 5 above)
Comment (cb-cs) [reply src/test/java/pdfverification/tests/HgvPRSBilingualTests.java:42]: done

File: src/test/java/pdfverification/tests/HgvPRSBilingualTests.java (hunk 6 above)
Comment (cb-cs) [reply src/test/java/pdfverification/tests/HgvPRSBilingualTests.java:19]: Yes that's fine i've no preference.

**Review by Daniel-Searle — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Pull request merged**